internal/yale/cache.go

```diff
@@ -0,0 +1,187 @@
+package yale
+
+import (
+	"context"
+	"fmt"
+	"github.com/broadinstitute/yale/internal/yale/cache"
+	"github.com/broadinstitute/yale/internal/yale/crd/api/v1beta1"
+	"github.com/broadinstitute/yale/internal/yale/logs"
+	"google.golang.org/api/iam/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+	"time"
+)
+
+// annotations added by Yale to its target secret
+const validAfterAnnotation = "validAfterDate"
+const currentServiceAccountKeyNameAnnotation = "serviceAccountKeyName"
+const oldServiceAccountKeyNameAnnotation = "oldServiceAccountKeyName"
+
+const timeInADay = time.Hour * 24
+
+func (m *Yale) PopulateCache() error {
+	logs.Info.Printf("Syncing Yale cache (namespace: %s)...", m.options.CacheNamespace)
+
+	if err := m.createCacheNamespaceIfMissing(); err != nil {
+		return err
+	}
+	if err := m.deleteAllCacheEntries(); err != nil {
+		return err
+	}
+	keysMap, err := m.getKeysMap()
+	if err != nil {
+		return err
+	}
+
+	for _, gcpSaKey := range keysMap {
+		secret, err := m.k8s.CoreV1().Secrets(gcpSaKey.Namespace).Get(context.Background(), gcpSaKey.Spec.Secret.Name, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("error retrieving secret for GCPSaKey %s/%s: %v", gcpSaKey.Namespace, gcpSaKey.Name, err)
+		}
+
+		if secret.Annotations[validAfterAnnotation] == "" {
```

Comment:
Secrets created by Yale have this annotation "validAfter" so I like this validation step.

Reply:
```suggestion
		if _, annotationExists := secret.Annotations[validAfterAnnotation]; annotationExists {
```
Just suggesting another way to perform this same check, this pattern seems to be common in other go code bases. Either works though

Reply:
I like that pattern generally but used `""` here because it guards against the annotation existing but having an empty value